code/xcvm/cosmwasm/contracts/gateway/src/contract/ibc/ics20.rs

```diff
@@ -35,68 +39,137 @@ pub(crate) fn handle_bridge_forward(
 	));
 
 	ensure_eq!(msg.msg.assets.0.len(), 1, ContractError::ProgramCannotBeHandledByDestination);
-	// algorithm to handle for multihop
-	// 1. recurse on program until can with memo
-	// 2. as soon as see no Spawn/Transfer, stop memo and do Wasm call with remaining Packet
-
 	let (local_asset, amount) = msg.msg.assets.0.get(0).expect("proved above");
 
-	let route: IbcIcs20Route = get_route(deps.storage, msg.to, *local_asset)?;
-	state::tracking::bridge_lock(deps.storage, (msg.clone(), route.clone()))?;
-
-	let asset = msg
-		.msg
-		.assets
-		.0
-		.get(0)
-		.map(|(_, amount)| (route.on_remote_asset, *amount))
-		.expect("not empty");
-
-	let packet = XcPacket {
-		interpreter: String::from(info.sender).into_bytes(),
-		user_origin: msg.interpreter_origin.user_origin.clone(),
-		salt: msg.msg.salt,
-		program: msg.msg.program,
-		assets: vec![asset].into(),
-	};
-
-	deps.api
-		.debug(&format!("cvm::gateway::ibc::ics20 route {}", &serde_json_wasm::to_string(&route)?));
-
-	let mut event = make_event("bridge")
-		.add_attribute("to_network_id", msg.to.to_string())
-		.add_attribute(
-			"assets",
-			serde_json_wasm::to_string(&packet.assets)
-				.map_err(|_| ContractError::FailedToSerialize)?,
-		)
-		.add_attribute(
-			"program",
-			serde_json_wasm::to_string(&packet.program)
-				.map_err(|_| ContractError::FailedToSerialize)?,
-		);
-	if !packet.salt.is_empty() {
-		let salt_attr = Binary::from(packet.salt.as_slice()).to_string();
-		event = event.add_attribute("salt", salt_attr);
-	}
-
-	let coin = Coin::new(amount.0, route.local_native_denom.clone());
+	let (msg, event) = if let Ok(transfer_shortcut) =
+		ibc_ics_20_transfer_shortcut(deps.as_ref(), &msg)
```

Comment:
Why are we ignoring error? As commented on the function, it seems to me that this function should return `Result<Option<_>, _>` and then this would be `if let Some(shortcut) = ibc_ics_20_transfer_shortcut(deps.as_ref(), &msg)?`.

Reply:
will be in next PR to act according shortcut. now no chains configured to use it.

```
unimplemented!("add tracking lock for funds return usual cosmos message to transfer as defined in {:?}", transfer_shortcut);
```